net/nimble/host/mesh/src/cfg.c

```diff
@@ -2913,6 +2913,8 @@ static void heartbeat_sub_set(struct bt_mesh_model *model,
 
 	if (period_ms) {
 		cfg->hb_sub.expiry = k_uptime_get() + period_ms;
+	} else {
+		cfg->hb_sub.expiry = k_uptime_get();
 	}
```

Comment:
you could just do
cfg->hb_sub.expiry = k_uptime_get() + period_ms;

without if-else

Reply:
shouldn't we set cfg->hb_sub.expiry = 0; instead?
@jhedberg  fyi

Reply:
Yes, expiry = 0 would be more appropriate

Reply:
Also note that this is not about invalid parameters. Messages with invalid parameters should not yield any reaction at all by the implementation. Instead, this particular code path is about disabling the subscription.